include/emul.h

```diff
@@ -0,0 +1,108 @@
+/*
```

Comment:
This file should probably be in include/drivers/emul.h given that the implementation is in drivers/emul

Reply:
> so maybe the implementation should move

yes, move it to subsys/emul maybe

Reply:
Yes, that  makes sense.

@sjg20 could you move the files from `drivers/emul/*` to `subsys/emul/i2c/` and change `eeprom_at24_emul.c` to `emul_atmel_at24.c`?  Rationale:
* This driver is specific to the i2c emulator API so should be a directory distinct from any future emulator APIs
* Globally available emulators should use the compatible name (`atmel,at24` in this case) as the pattern for their primary identifier.  The Kconfig prefix should match: `EMUL_I2C_ATMEL_AT24`.

I don't think it's necessary to make big changes to `drivers/i2c_emul` because that actually is an I2C driver for emulators.  It's in the right place, but for consistency with the compatible and header the implementation file would be better named `i2c_emul.c`.